src/edgekv/ekv-service.ts

```diff
@@ -38,32 +39,55 @@ export function getInitializedEdgeKV() {
+  let writeItemFromJsonpath = `${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}/groups/${groupId}/items/${itemId}`; 
+  if(sandboxid){
+    writeItemFromJsonpath += `?sandboxId=${sandboxid}`
+  }
+  return httpEdge.sendEdgeRequest(writeItemFromJsonpath, 'PUT', fs.readFileSync(itemPath, { encoding: null}).toString('utf8'), {
     'Content-Type': 'application/json'
   }, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.writeItem).then(r => r.body).catch(err => error.handleError(err));
 }
 
-export function readItem(network: string, namespace: string, groupId: string, itemId: string) {
-  return httpEdge.getJson(`${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}/groups/${groupId}/items/${itemId}`, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.readItem).then(r => r.body).catch(err => error.handleError(err));
+export function readItem(network: string, namespace: string, groupId: string, itemId: string,sandboxid:string) {
+  let readItemPath = `${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}/groups/${groupId}/items/${itemId}`;
+  if(sandboxid){
+    readItemPath += `?sandboxId=${sandboxid}`;
+  }
+  return httpEdge.getJson(readItemPath, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),ekvMetrics.readItem).then(r => r.body).catch(err => error.handleError(err));
 }
 
-export function deleteItem(network: string, namespace: string, groupId: string, itemId: string) {
-  return httpEdge.deleteReq(`${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}/groups/${groupId}/items/${itemId}`, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.deleteItem).then(r => r.body).catch(err => error.handleError(err));
+export function deleteItem(network: string, namespace: string, groupId: string, itemId: string,sandboxid:string) {
+  let deleteItemPath = `${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}/groups/${groupId}/items/${itemId}`;
+  if(sandboxid){
+    deleteItemPath += `?sandboxId=${sandboxid}`;
+  }
+  return httpEdge.deleteReq(deleteItemPath, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),ekvMetrics.deleteItem).then(r => r.body).catch(err => error.handleError(err));
 }
 
-export function getItemsFromGroup(network: string, namespace: string, groupId: string, maxItems: number) {
+export function getItemsFromGroup(network: string, namespace: string, groupId: string, maxItems: number,sandboxid:string) {
   var qs: string = "";
   if (maxItems !== undefined) {
     qs += `?maxItems=${maxItems}`;
   }
-  return httpEdge.getJson(`${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}/groups/${groupId}${qs}`, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.readItemsFromGroup).then(r => r.body).catch(err => error.handleError(err));
+  let listItemsPath = `${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}/groups/${groupId}${qs}`;
+  if(sandboxid){
+    let ls:string = "";
+    if(maxItems){
+      ls += "&";
+    }
+    listItemsPath += `${ls}?sandboxId=${sandboxid}`;
```

Comment:
this would be a bug. If maxItems is present we are constructing a query parameter like "staging/namespaces/ns1/groups/grp1?maxItems=2**&?**sandboxId=sid1. Its appending both &? if maxitems is present.

Reply:
when I tested API its working .

% echo "sandbox_write_item_test" | http --auth-type edgegrid -a sqa-shared: GET ":/edgekv/v1/networks/staging/namespaces/default/groups/group1?maxItems=2&?sandboxId=dc0e0d67-28cc-4276-82cc-c88c1a2d8cfd" Content-Type:application/json accountSwitchKey==B-M-28QYF3M
HTTP/1.1 200 OK
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Connection: keep-alive
Content-Encoding: gzip
Content-Length: 34
Content-Type: application/json;charset=utf-8
Date: Fri, 25 Feb 2022 21:22:17 GMT
Expires: 0
Pragma: no-cache
Server: nginx
Vary: accept-encoding
x-content-type-options: nosniff
x-frame-options: DENY
x-ids-session-id: 4e349d5a-ae83-4c03-93e7-6c487fc48398
x-trace-id: da4e621948873dc5
x-xss-protection: 1; mode=block

[
    "yes",
    "key1"
]